kong/plugins/acme/client.lua

```diff
@@ -524,9 +549,29 @@ local function renew_certificate(premature)
 
     if plugin.name == "acme" then
       kong.log.info("renew storage configured in acme plugin: ", plugin.id)
+
+      -- If Renew Worker Key Not Exists
+      local worker, _ = get_worker_config(plugin.config)
+      if worker then
+        kong.log.notice("Worker lock exists in acme storage. Another worker '" .. worker .. "' seems to be renewing. Skipping renewal flow for worker " .. ngx.worker.id())
+        return
+      end
+      --  Create & Save Worker Key
+      local err = store_worker_config(plugin.config)
```

Comment:
seems multiple workers could successfully set the config and continue. don't we need to lock?

Reply:
implemented dao add(), which returns error if config object already exists in db 
🔗 https://github.com/Kong/kong-plugin-acme/blob/f236fac0abdd76c952eb6cd34c8780d36d25074b/kong/plugins/acme/storage/kong.lua#L16